### PR TITLE
[mdns]: Add linux console functional tests

### PR DESCRIPTION
**Pull request by david-cermak**
## Adds a simple mdns host test suite

* running on host
* using (vanilla) pytest
* using `dnspython` library to check mdns records

## Future considerations

- ~~Using idf container vs. vanilla ubuntu image? (idf containers are tailored to build-only jobs, but we need to run host tests as well; used ubuntu-20.04 image, as there're some issues with running dig command inside 22.04 container)~~
- Replace https://github.com/espressif/esp-protocols/tree/master/components/mdns/tests/test_apps with this test app (run the same test with pytest embedded on the real target)

### CODE REVIEW

File: components/mdns/tests/host_test/pytest_mdns.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+    def terminate(self):
+        self.send_input('exit')
+        self.get_output('Exit')
+        self.process.wait()
+        self.process.close()
+        assert self.process.exitstatus == 0
+
+
+class DigWrapper:
```
[the remark this answers is not in the document]
Comment (david-cermak) [reply components/mdns/tests/host_test/pytest_mdns.py:40]: Yes, `dnspython` was my first choice, but it doesn't provide a very verbose output/results (I'd like to verify both the question and answer fields, which `dig` provides). Alternatively we can use the raw dns packet library in python and implement the resolver, but that would be more work.

File: components/mdns/tests/host_test/sdkconfig.defaults [hunk 2]
```diff
@@ -1,9 +1,6 @@
-CONFIG_IDF_TARGET="linux"
+CONFIG_TEST_NETIF_NAME="eth0"
```
[the remark this answers is not in the document]
Comment (david-cermak) [reply components/mdns/tests/host_test/sdkconfig.defaults:1]: > Shouldn't the IDF_TARGET config stay?

no, wanted this project working on real target as well.

**Review by euripedesrocha — APPROVED** [review]

**Review by euripedesrocha — COMMENTED** [review]
Just a minor comment.

File: .github/workflows/mdns__host-tests.yml [hunk 3]
```diff
@@ -10,27 +10,30 @@ on:
...
 
     steps:
       - name: Checkout esp-protocols
         uses: actions/checkout@v4
         with:
-          path: esp-protocols
+          path: protocols
 
-      - name: Build and Test
+      - name: Build tests
```
[the remark this answers is not in the document]
Comment (david-cermak) [reply .github/workflows/mdns__host-tests.yml]: thanks, fixed!

**Pull request merged**